src/docs-developer-guide.rst

```diff
@@ -122,3 +122,188 @@ For example:
   }
 
   /* New section ends */
+
+.. _style-tests:
+
+Style Guide checks
+--------------------
+
+Proselint_  is used for style testing the docs.
+Apart from the inbuilt tests in proselint,
+custom checks are added for style guide testing.
+We follow the approach of `literate programming <https://en.wikipedia.org/wiki/Literate_programming>`_.
+The checks reside in the file :doc:`docs-style-guide.rst <docs-style-guide>`.
+After each style rule,
+you can define a python code-block
+containing the code for style testing.
+When style testing is performed,
+these python code-blocks are parsed to
+generate a testing script.
+
+.. _proselint-checks:
+
+Proselint dependent checks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```

Comment:
This section does not make clear that to write a style check,
you write a new function that calls one of the built-in stylecheck functions as return value.
One could figure this out from looking at the example at the end (or by looking at the code in style-guide)
but it would be more clear to lay this out at the beginning.